app/controllers/pings_controller.rb

```diff
@@ -18,7 +18,32 @@ def index
     if !pings.empty?
       render json: pings, each_serializer: PingIndexSerializer
     else
-      render json: { message: 'Unfortunately no one has planned to go shopping, so maybe you can?' }
+      render json: {
+               message:
+                 'Unfortunately no one has planned to go shopping, so maybe you can?'
+             }
+    end
+  end
+
+  def update
+    requested_ping = Ping.all.find(params[:id])
+    if requested_ping.user_id == current_user.id
+      requested_ping.update(active: false)
+      render json: { message: 'Your trip is now closed for more requests' }
+    else
+      render json: {
+               message: "You are not authorized to update another user's ping"
+             },
+             status: 401
+    end
```

Comment:
same here with the formatting, not sure if its what the code looks like in vs code